pkg/cmd/operator/ignition.go

```diff
@@ -0,0 +1,375 @@
+		if len(serviceNameValidationErrs) != 0 {
+			errs = append(errs, fmt.Errorf("invalid service name %q: %v", o.ServiceName, serviceNameValidationErrs))
+		}
+	}
+
+	if !slices.ContainsItem(validation.SupportedBroadcastAddressTypes, scyllav1.BroadcastAddressType(o.NodesBroadcastAddressTypeString)) {
+		errs = append(errs, fmt.Errorf("unsupported value of nodes-broadcast-address-type %q, supported ones are: %v", o.NodesBroadcastAddressTypeString, validation.SupportedBroadcastAddressTypes))
+	}
+
+	if !slices.ContainsItem(validation.SupportedBroadcastAddressTypes, scyllav1.BroadcastAddressType(o.ClientsBroadcastAddressTypeString)) {
+		errs = append(errs, fmt.Errorf("unsupported value of clients-broadcast-address-type %q, supported ones are: %v", o.ClientsBroadcastAddressTypeString, validation.SupportedBroadcastAddressTypes))
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (o *IgnitionOptions) Complete(args []string) error {
+	err := o.ClientConfig.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = o.InClusterReflection.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = o.ServeProbesOptions.Complete(args)
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.ProtoConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %w", err)
+	}
+
+	o.nodesBroadcastAddressType = scyllav1.BroadcastAddressType(o.NodesBroadcastAddressTypeString)
+	o.clientsBroadcastAddressType = scyllav1.BroadcastAddressType(o.ClientsBroadcastAddressTypeString)
+
+	return nil
+}
+
+func (o *IgnitionOptions) Run(originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+
+	return o.RunCtx(ctx, originalStreams, cmd)
+}
+
+func (o *IgnitionOptions) RunCtx(ctx context.Context, originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	startTime := time.Now()
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	healthzFunc := func(w http.ResponseWriter, r *http.Request) {
+		ignited := o.ignited.Load()
+		klog.V(3).InfoS("Probe call", "URI", r.RequestURI, "Ignited", ignited)
+		if !ignited {
+			w.WriteHeader(http.StatusServiceUnavailable)
+		}
+		return
+	}
+	o.mux.HandleFunc(naming.ReadinessProbePath, healthzFunc)
+	o.mux.HandleFunc(naming.LivenessProbePath, healthzFunc)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		err := o.ServeProbesOptions.Execute(ctx, originalStreams, cmd)
+		if err != nil {
+			klog.ErrorS(err, "Error running probe server")
+		}
+	}()
+
+	// Wait for the service that holds identity for this scylla node.
+	klog.InfoS("Waiting for Service availability and IP address",
+		"Service", naming.ManualRef(o.Namespace, o.ServiceName),
+	)
```

Comment:
nit: inconsistent with codebase and unnecessary line break, line isn't very long